src/client/datascience/data-viewing/data-wrangler/dataWrangler.ts

```diff
@@ -0,0 +1,502 @@
+                transformation: DataScience.dataWranglerImportTransformation(),
+                code: DataScience.dataWranglerImportCode().format(this.sourceFile ?? 'broken'),
+                variableName: 'df'
+            });
+            this.postMessage(DataWranglerMessages.UpdateHistoryList, this.historyList).ignoreErrors();
+        }
+    }
+
+    private dataWranglerDisposed() {
+        this._onDidDisposeDataWrangler.fire(this as IDataWrangler);
+    }
+
+    public async updateWithNewVariable(newVariableName: string) {
+        const notebook = (this.dataProvider as IJupyterVariableDataProvider).notebook;
+
+        // Generate a variable
+        const jupyterVariable = await this.kernelVariableProvider.getFullVariable(
+            {
+                name: newVariableName,
+                value: '',
+                supportsDataExplorer: true,
+                type: 'DataFrame',
+                size: 0,
+                shape: '',
+                count: 0,
+                truncated: true
+            },
+            notebook
+        );
+        const jupyterVariableDataProvider = await this.dataProviderFactory.create(jupyterVariable);
+        // Set dependencies for jupyterVariableDataProvider
+        jupyterVariableDataProvider.setDependencies(jupyterVariable, notebook);
+        // Get variable info
+        this.dataFrameInfoPromise = jupyterVariableDataProvider.getDataFrameInfo();
+        this.dataProvider = jupyterVariableDataProvider;
+        const dataFrameInfo = await this.dataFrameInfoPromise;
+        super.setTitle(`Data Wrangler`);
+
+        this.postMessage(DataViewerMessages.InitializeData, dataFrameInfo).ignoreErrors();
+    }
+
+    public async getHistoryItem(index: number) {
+        const variableName = this.historyList[index].variableName;
+        await this.updateWithNewVariable(variableName);
+    }
+
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    protected onMessage(message: string, payload: any) {
+        switch (message) {
+            case DataWranglerMessages.SubmitCommand:
+                this.handleCommand(payload).ignoreErrors();
+                break;
+
+            case DataWranglerMessages.RefreshDataWrangler:
+                this.refreshData().ignoreErrors();
+                break;
+
+            case InteractiveWindowMessages.LoadTmLanguageRequest:
+                void this.requestTmLanguage(payload);
+                break;
+
+            case InteractiveWindowMessages.LoadOnigasmAssemblyRequest:
+                void this.requestOnigasm();
+                break;
+
+            case CssMessages.GetMonacoThemeRequest:
+                void this.handleMonacoThemeRequest(payload);
+                break;
+
+            default:
+                break;
+        }
+
+        // Some messages will be handled by DataViewer
+        super.onMessage(message, payload);
+    }
+
+    private async requestTmLanguage(languageId: string = PYTHON_LANGUAGE) {
```

Comment:
We must be doing this somewhere else? Ah yes we are. InteractiveBase. I think you can move this up one level to WebviewPanelHost and then that would work for this code too.

Then all of the monaco init stuff will be in one spot.

Reply:
Meaning change InteractiveBase and this code to not have these monaco related functions, but instead move them to WebviewPanelHost.

Reply:
The IMapping part is a little tricky, I'll message you about it.